src/pages/Cart/AsideCard.tsx

```diff
@@ -0,0 +1,172 @@
+                )}
+              </p>
+            </div>
+            {Array.from(discounts.entries()).map(([key, value], index) => (
+              <div
+                key={key}
+                className="flex items-stretch justify-between text-sm"
+              >
+                <p className="text-default-300">{key}</p>
+                <div className="inline-flex gap-2">
+                  <p className="text-default-300">
+                    -
+                    {formatPrice(
+                      cart.discountOnTotalPrice?.includedDiscounts[index]
+                        .discountedAmount.centAmount || 0,
+                    )}
+                  </p>
+                  <button
+                    className="flex size-[20px] items-center justify-center rounded-full border-2 hover:border-slate-400"
+                    onClick={() => {
+                      cancelDiscountById(value.id).then((response) => {
+                        if (response) {
+                          setDiscounts(new Map());
+                        }
+                      });
+                    }}
+                  >
+                    <X size={10} strokeWidth={3} />
+                  </button>
+                </div>
+              </div>
+            ))}
+            <Divider className="my-1" />
+          </>
+        )}
+        <div className="flex content-stretch gap-3">
+          <p className="inline-block text-default-500">
+            {cart.discountOnTotalPrice ? 'Subtotal' : 'Total'}
+          </p>
+          <p className="inline-block  text-black dark:text-white">
+            {formatPrice(cart.totalPrice.centAmount)}
+          </p>
+        </div>
+      </div>
+      <Divider />
+      <CardBody className="gap-2">
+        <p className="font-thin text-red-500">{error ? error : ''}</p>
+        <Input
+          defaultValue={''}
+          endContent={
+            code ? (
+              <Button
+                isIconOnly
+                className="w-[50px]  uppercase"
+                color="success"
+                radius="full"
+                variant="flat"
+                onClick={() => applyDiscounts(code).finally(() => setCode(''))}
+              >
+                <Check strokeWidth={3} />
+              </Button>
+            ) : undefined
+          }
+          label="Promo Code"
+          type="text"
+          value={code}
+          variant="bordered"
+          onValueChange={(value: string) => setCode(value)}
+        />
+      </CardBody>
+      <Divider />
+      <CardFooter>
+        <PopoverCart
+          action={() => clearCart()}
+          isLoading={isLoading}
+          triggerButton={
+            <Button
+              className="w-[200px] font-bold uppercase"
+              color="warning"
+              radius="full"
+              variant="light"
+            >
+              Clear shopping cart
+            </Button>
+          }
+        />
+      </CardFooter>
+    </Card>
```

Comment:
better to split big jsx like this into components. Array of discounts in this jsx can be moved to separate component